std/shared/src/main/scala/cats/effect/std/Hotswap.scala

```diff
@@ -131,12 +131,12 @@ object Hotswap {
         new Hotswap[F, R] {
 
           override def swap(next: Resource[F, R]): F[R] =
-            exclusive.surround {
-              F.uncancelable { poll =>
-                poll(next.allocated).flatMap {
-                  case (r, fin) =>
-                    swapFinalizer(Acquired(r, fin)).as(r)
-                }
+            F.uncancelable { poll =>
+              poll(next.allocated).flatMap {
+                case (r, fin) =>
+                  poll(exclusive.onCancel(Resource.eval(fin)).surround {
+                    swapFinalizer(Acquired(r, fin)).uncancelable.as(r)
```

Comment:
I think there is a cancelation bug here. If we are canceled after `exclusive.onCancel(...)` and before `swapFinalizer` then `fin` is lost. I just push a test in d528afbd95078ba7e8c55ef79efbe3095cbbf63f that demonstrates the problem.

Reply:
I wonder if something like this can work.

```suggestion
                  exclusive.mapK(poll).onCancel(Resource.eval(fin)).surround {
                    swapFinalizer(Acquired(r, fin)).uncancelable.as(r)
```